app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -90,13 +94,42 @@ class PatientSearchResultItemView(
     addressLabel.text = address.completeAddress
   }
 
-  private fun renderPatientNameAgeAndGender(fullName: String, gender: Gender, dateOfBirth: DateOfBirth) {
+  private fun highlightPatientName(searchQuery: String?, patientSearchResult: PatientSearchResultViewModel, dateOfBirth: DateOfBirth): Name {
+    val canHighlight = searchQuery?.let { searchQuery.isNotBlank() && patientSearchResult.fullName.contains(it, ignoreCase = true) } == true
+    return if (canHighlight) {
+      Name.Highlighted(
+          patientName = showPatientNameAgeAndGender(patientSearchResult.fullName, patientSearchResult.gender, dateOfBirth),
```

Comment:
We are calling `showPatientNameAgeAndGender` irrespective of whether it's highlighted or not. Let's extract it out as a variable inside this function. Also, do we need to have it as a separate function considering it's just getting string resources?

Reply:
I think yes, we already are doing something in this function. Keeping a different function for this makes sense. WDYT?

Reply:
What we are doing is creating the patient name. So it makes sense to include it in this function rather than having it as a separate function. As long as it doesn't trigger a code climate issue we can include getting strings in this function.

Reply:
So we're not creating patient name, instead checking whether the patient name is supposed to be highlighted or not. Setting the patient name into a different function just looks for readable to me.

Reply:
We are constructing `Name.Highlighted` and `Name.Plain` here, right? and we also need to pass in the string of patient name, age and gender to those classes. Doesn't that mean we constructing/creating the patient name here? Even if we want to leave it as a separate function then `showPatientNameAgeAndGender` is getting the strings from resources, the function name implies that we are displaying the text.